source/draw/gpu/opengl/Map/OpenGLMap.ooc

```diff
@@ -65,23 +65,23 @@ OpenGLMap: abstract class extends GpuMap {
 				cell := value as Cell
 				match (cell T) {
 					case Int => program setUniform(key, cell as Cell<Int> get())
-					case IntPoint2D => program setUniform(key, cell as Cell<IntPoint2D> get())
-					case IntPoint3D => program setUniform(key, cell as Cell<IntPoint3D> get())
-					case IntSize2D => program setUniform(key, cell as Cell<IntSize2D> get())
-					case IntSize3D => program setUniform(key, cell as Cell<IntSize3D> get())
+					case IntPoint2D => point := cell as Cell<IntPoint2D> get(); program setUniform(key, point x, point y)
+					case IntPoint3D => point := cell as Cell<IntPoint3D> get(); program setUniform(key, point x, point y, point z)
+					case IntSize2D => size := cell as Cell<IntSize2D> get(); program setUniform(key, size width, size height)
+					case IntSize3D => size := cell as Cell<IntSize3D> get(); program setUniform(key, size width, size height, size depth)
 					case Float => program setUniform(key, cell as Cell<Float> get())
-					case FloatPoint2D => program setUniform(key, cell as Cell<FloatPoint2D> get())
-					case FloatPoint3D => program setUniform(key, cell as Cell<FloatPoint3D> get())
-					case FloatPoint4D => program setUniform(key, cell as Cell<FloatPoint4D> get())
-					case FloatSize2D => program setUniform(key, cell as Cell<FloatSize2D> get())
-					case FloatSize3D => program setUniform(key, cell as Cell<FloatSize3D> get())
+					case FloatPoint2D => point := cell as Cell<FloatPoint2D> get(); program setUniform(key, point x, point y)
+					case FloatPoint3D => point := cell as Cell<FloatPoint3D> get(); program setUniform(key, point x, point y, point z)
+					case FloatPoint4D => point := cell as Cell<FloatPoint4D> get(); program setUniform(key, point x, point y, point z, point w)
+					case FloatSize2D => size := cell as Cell<FloatSize2D> get(); program setUniform(key, size width, size height)
+					case FloatSize3D => size := cell as Cell<FloatSize3D> get(); program setUniform(key, size width, size height, size depth)
 					case FloatTransform2D => program setUniform(key, cell as Cell<FloatTransform2D> get())
 					case FloatTransform3D => program setUniform(key, cell as Cell<FloatTransform3D> get())
-					case ColorBgr => program setUniform(key, cell as Cell<ColorBgr> get())
-					case ColorBgra => program setUniform(key, cell as Cell<ColorBgra> get())
-					case ColorUv => program setUniform(key, cell as Cell<ColorUv> get())
-					case ColorYuv => program setUniform(key, cell as Cell<ColorYuv> get())
-					case ColorYuva => program setUniform(key, cell as Cell<ColorYuva> get())
+					case ColorBgr => color := cell as Cell<ColorBgr> get(); program setUniform(key, color blue as Float / 255, color green as Float / 255, color red as Float / 255)
+					case ColorBgra => color := cell as Cell<ColorBgra> get(); program setUniform(key, color blue as Float / 255, color green as Float / 255, color red as Float / 255, color alpha as Float / 255)
```

Comment:
Color vectors in GLSL (excluding texture packing) are always in RGBA order so it makes sense to reverse them here to prevent the red channel from being in color.b and blue channel from being in color.r.